lib/microos.pm

```diff
@@ -22,7 +22,9 @@ our @EXPORT = qw(microos_reboot microos_login);
 
 # Assert login prompt and login as root
 sub microos_login {
-    assert_screen 'linux-login-microos', 150;
+    unless (match_has_tag('linux-login-microos') || match_has_tag('linux-login')) {
```

Comment:
I don't get the differentiation between `linux-login-microos` and `linux-login` in needles, AFAICT it's all the same and there can't really be any conflict. But that's a topic for some other time...

In this particular case I wonder why it's fine with `linux-login` as already matched tag, but doesn't accept it in the `assert_screen` here or in below in line 52.

Reply:
The reason for adding `linux-login` in the condition is to avoid matching the same stuff twice and save 1 step. For example [here](https://openqa.suse.de/tests/4834939#step/disk_boot/2)  matches first `linux-login` and in the next frame it matches `linux-login-microos`.

Maybe I can just skip that condition if it's more clear, as it works anyway.

Reply:
It sounds like whatever matched `linux-login` there should also match `linux-login-microos`, then the second `match_has_tag` here could be dropped again and it's consistenly `linux-login-microos` throughout.

Reply:
AFAICT it's from `wait_boot_textmode`

Reply:
> AFAICT it's from `wait_boot_textmode`
Yes, `disk_boot` calls `wait_boot`, which calls `wait_boot_past_bootloader` and `wait_boot_textmode`. This will match `linux-login` needle and then we are also forcing `linux-login-microos` as well...

See this error if I remove it from the condition as it is now:  http://fromm.arch.suse.de/tests/487#

Reply:
Yeah, either `linux-login-microos` should also have the `linux-login` tag or every use of `linux-login` should also match `linux-login-microos` there. This particular failure looks like either a product bug (screen not cleared) or a missing needle for the case that `login:` is at the bottom.

Reply:
yeah, btw, the `linux-login-microos` needle already has the tag `linux-login`.

Reply:
Not here (openSUSE):

```
/var/lib/openqa/tests/opensuse/products/opensuse/needles> git grep linux-login *.json
autoyast-system-login-console-minimal-20181012.json:    "linux-login"
linux-login-20140916.json:    "linux-login"
linux-login-20200525.json:    "linux-login"
linux-login-autoyast-20150724.json:    "linux-login"
linux-login-autoyast-20160822.json:    "linux-login"
linux-login-autoyast-20191126.json:    "linux-login",
linux-login-autoyast-20200203.json:    "linux-login"
linux-login-kubic-20180201.json:    "linux-login-microos"
linux-login-kubic-20190408.json:    "linux-login-microos"
linux-login-kubic-20190415.json:    "linux-login-microos"
linux-login-kubic-20191031.json:    "linux-login-microos"
linux-login-kubic-20200428.json:    "linux-login-microos"
linux-login-kubic-20200909.json:    "linux-login-microos"
```

Reply:
my bad, in Suse either. I'll try to fix that then adding that tag as well...

Reply:
Thanks, then `linux-login-microos` can be replaced everywhere, which should make this simpler and more consistent as well. Might even make some needles redundant, resulting in quicker matching